server/src/core/agent/cache.py

```diff
@@ -1,21 +1,26 @@
-from core.common.encode import encode_to_base64, decode_from_base64
-from core.common.cache import CacheMemory
+# from core.common.encode import encode_to_base64, decode_from_base64
+from core.common import conn
 
 
+# TODO: Why were there used base64 encodings?
+#       Experiments show that it receives binary strings as arguments and works fine.
 class PredictCache:
     def __init__(self):
-        self.cache = CacheMemory(30)
+        # self.cache = CacheMemory(30)  # TODO: Limit the number of cached entries?
+        self.redis = conn.get_redis_instance()  # TODO: Rewrite with dependency injection.
```

Comment:
The entries limit was required to protect the memory. It is not necessary anymore

Make sure the redis instace is passed in the constructor